ea_restaurant_data_loader/lib/utils/oauth2_util.ex

```diff
@@ -0,0 +1,21 @@
+defmodule EaRestaurantDataLoader.Oauth2Util do
+  alias EaRestaurantDataLoader.Token
+
+  defp signer(secret_key) do
+    Joken.Signer.create("HS256", secret_key)
+  end
+
+  def build_client_credentials_token(client_name, scopes, exp_time, secret_key) do
+    extra_claims = %{
+      "exp" => DateTime.utc_now() |> DateTime.add(exp_time, :second) |> DateTime.to_unix(),
+      "clientName" => client_name,
+      "scopes" => scopes
+    }
+
+    token = Token.generate_and_sign!(extra_claims, signer(secret_key))
+  end
+
+  def validate_token(token, secret_key) do
+    {:ok, claims} = Token.verify_and_validate(token, signer(secret_key))
```

Comment:
wondering if it is better to create a `decode_token/2` function? Can you check how do you handle the validate method in the other projects?

Reply:
This is a similiar aprouch as python where same function validate and decode.

![Capture](https://github.com/eapg/ea_restaurant_data_loader/assets/77887982/5a700fc6-b572-4e40-9d8d-d4c5cc8eb875)

Reply:
yeah but looks like we don't return anything from here

Reply:
Looking deeper in the python project in the moment where i need the claims i call `jwt.decode()` directly. but in this case is like create two functions with same commands, one that return the claims and other that don't return anything, only throw an error when the token is not valid. I think that create two functions that call `Token.verify_and_validate()` is redundant.

Reply:
Done.